core/src/main/java/org/fao/geonet/api/records/attachments/AbstractStore.java

```diff
@@ -206,6 +206,16 @@ public MetadataResource patchResourceStatus(final ServiceContext context, final
         return patchResourceStatus(context, metadataUuid, resourceId, metadataResourceVisibility, true);
     }
 
+    @Override
+    public void copyResources(ServiceContext context, String sourceUuid, String targetUuid, MetadataResourceVisibility metadataResourceVisibility, boolean sourceApproved, boolean targetApproved) throws Exception {
+        final List<MetadataResource> resources = getResources(context, sourceUuid, metadataResourceVisibility, null, true);
```

Comment:
I believe "true" should be using variable sourceApproved

Reply:
yes. This causes @josegar74  test failure. That's why when he copies from working copy -> approved copy. The files always got wiped out because this hardcoded flag "true" always gets sources as approved copy. So the File.copy method copies the input stream of the same file to the same location and causing issues. Now it should get fixed.